model_compression_toolkit/core/common/mixed_precision/resource_utilization_tools/resource_utilization_data.py

```diff
@@ -132,6 +133,54 @@ def compute_nodes_weights_params(graph: Graph, fw_info: FrameworkInfo) -> Tuple[
 
     return np.array(weights_memory_bytes), np.array(weights_params)
 
+
+def compute_activation_output_maxcut_sizes(graph: Graph) -> Tuple[np.ndarray, np.ndarray]:
```

Comment:
This is eventually supposed to be removed and replaced with a single implementation of the ru metrics.
so no need to further optimize this code,
but @irenaby please take a look to make sure that this can be integrated into your change to ru data